### ISSUE TITLE
Tech Debt: remove secondary stack check

### ISSUE THREAD

**Issue by Ekleog-NEAR** [near/wasmer#153]
At some point we introduced a secondary stack check into wasmer itself (see `update_max_stack_depth` & co)

Once we land the finite-wasm work, we should probably remove this secondary stack check.

**Comment by nagisa** [comment]
Somewhat a duplicate of https://github.com/near/wasmer/issues/130.

**Comment by Ekleog-NEAR** [comment]
Oh I missed it, just from the title I hadn’t guessed it was related.

That said I think the end-goal is somewhat different, and so it might make sense to keep both open? (eg. I think after landing finite-wasm into wasmer we could probably handle this one and remove some code, but #130 needs more in-depth changes to clean things up properly)
